web/src/main/bvmui/src/views/Ledgers.vue

```diff
@@ -93,24 +134,43 @@ export default {
             showLedgerMetadata: false,
             currentLedger: null,
             ledgers: [],
+            ledgersCount: 0,
             totalSize: 0
         };
     },
-    created() {
-        let url = "api/ledger/all";
-        if (this.$route.meta.type === "bookie") {
-            const { bookieId, clusterId } = this.$route.params;
-            url = "api/ledger/all?bookie=" + encodeURIComponent(bookieId)
-                + "&cluster=" + encodeURIComponent(clusterId);
+    async created() {
+        return this.refreshLedgers(1, DefaultPageSize);
+    },
+    computed: {
+        pageLength() {
+            return Math.ceil(this.ledgersCount / this.size);
         }
-        this.$request.get(url).then(
-            ledgersResult => {
-                this.ledgers = ledgersResult.ledgers;
-                this.totalSize = ledgersResult.totalSize;
-            }
-        );
     },
     methods: {
+        async refreshLedgers(page, size) {
+            this.page = this.size === size ? page : 1;
+            this.size = size;
+            let url = `api/ledger/all?page=${this.page}&size=${this.size}`;
+            if (this.search) {
+                url += "&term=" + encodeURIComponent(this.searchTerm)
```

Comment:
may be better to create an utility method for creating safe query strings, there are lightweight libraries such as "querystring"

Reply:
Done, much better, thank you